crates/dojo-bindgen/src/plugins/typescript_v2/mod.rs

```diff
@@ -0,0 +1,594 @@
+        {system_initializations}
+    }}
+
+    async findEntities<T extends Query>(query: T, limit = 10, offset = 0) {{
+        const torii = await this.toriiPromise;
+
+        const clause = convertQueryToToriiClause(query);
+
+        const toriiResult = await torii.getEntities({{
+            limit,
+            offset,
+            clause,
+        }});
+
+        const result = Object.values(toriiResult).map((entity: any) => {{
+            return extractQueryFromResult<Query>(
+                query,
+                entity
+            ) as QueryResult<T>;
+        }});
+
+        return result as QueryResult<T>[];
+    }}
+
+    async findEntity<T extends Query>(query: T) {{
+        const result = await this.findEntities(query, 1);
+
+        if (result.length === 0) {{
+            return undefined;
+        }}
+
+        return result[0] as QueryResult<T>;
+    }}
+}}"
+        );
+
+        out
+    }
+
+    // Token should be a struct
+    // This will be formatted into a TypeScript interface
+    // using TypeScript defined types
+    fn format_struct(token: &Composite, handled_tokens: &[Composite]) -> String {
+        let mut native_fields: Vec<String> = Vec::new();
+
+        for field in &token.inners {
+            let mapped = TypeScriptV2Plugin::map_type(field.token.type_name().as_str());
+            if mapped == field.token.type_name() {
+                let token = handled_tokens
+                    .iter()
+                    .find(|t| t.type_name() == field.token.type_name())
+                    .unwrap_or_else(|| panic!("Token not found: {}", field.token.type_name()));
+                if token.r#type == CompositeType::Enum {
+                    native_fields.push(format!("{}: {};", field.name, mapped));
+                } else {
+                    native_fields.push(format!("{}: {};", field.name, field.token.type_name()));
+                }
+            } else {
+                native_fields.push(format!("{}: {};", field.name, mapped));
+            }
+        }
+
+        format!(
+            "
+// Type definition for `{path}` struct
+export interface {name} {{
+    {native_fields}
+}}
+",
+            path = token.type_path,
+            name = token.type_name(),
+            native_fields = native_fields.join("\n    ")
+        )
+    }
+
+    // Token should be an enum
+    // This will be formatted into a C# enum
+    // Enum is mapped using index of cairo enum
+    fn format_enum(token: &Composite) -> String {
+        let fields = token
+            .inners
+            .iter()
+            .map(|field| format!("{},", field.name,))
+            .collect::<Vec<String>>()
+            .join("\n    ");
+
+        format!(
+            "
+// Type definition for `{}` enum
+export enum {} {{
+    {}
+}}
+",
+            token.type_path,
+            token.type_name(),
+            fields
+        )
+    }
+
+    // Formats a system into a C# method used by the contract class
```

Comment:
```suggestion
    // Formats a system into a JS method used by the contract class
```
re-word to match the JS case. Other part may be adjusted.